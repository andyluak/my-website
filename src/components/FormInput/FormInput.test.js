import React from 'react';
import { shallow, render } from 'enzyme';
import FormInput from './FormInput';

describe( 'FormInput component tests', () =>{
	const mockHandleChange = jest.fn();
	const mockLabel = 'Test Label';
	const mockOtherProps = {
		name: 'test',
		type: 'text',
		value: 'test',
	};

	it( 'excepct to render FormInput component', () => {
		expect(shallow(<FormInput label={mockLabel} handleChange={mockHandleChange} {...mockOtherProps} />)).toMatchSnapshot();
	})

	// expect to render with value
	it( 'excepct to render FormInput component with value', () => {
		expect(render(<FormInput label={mockLabel} handleChange={mockHandleChange} {...mockOtherProps} value={1}/>)).toMatchSnapshot();
	})
})


