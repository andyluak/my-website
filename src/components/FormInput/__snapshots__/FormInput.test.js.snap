// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component tests excepct to render FormInput component 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    name="test"
    onChange={[MockFunction]}
    type="text"
    value="test"
  />
  <label
    className="shrink form-input-label"
  >
    Test Label
  </label>
</div>
`;

exports[`FormInput component tests excepct to render FormInput component with value 1`] = `
<div
  class="group"
>
  <input
    class="form-input"
    name="test"
    type="text"
    value="1"
  />
  <label
    class=" form-input-label"
  >
    Test Label
  </label>
</div>
`;
